a10_octavia/controller/worker/tasks/a10_database_tasks.py

```diff
@@ -218,11 +218,16 @@ def execute(self, loadbalancer):
 class GetBackupVThunderByLoadBalancer(BaseDatabaseTask):
     """ Get VThunder details from LoadBalancer"""
 
-    def execute(self, loadbalancer):
+    def execute(self, loadbalancer, vthunder=None):
         loadbalancer_id = loadbalancer.id
-        vthunder = self.vthunder_repo.get_backup_vthunder_from_lb(
+        backup_vthunder = self.vthunder_repo.get_backup_vthunder_from_lb(
             db_apis.get_session(), loadbalancer_id)
-        return vthunder
+
+        # VCS vMaster/vBlade may switched
+        if vthunder and backup_vthunder.ip_address == vthunder.ip_address:
+            backup_vthunder = self.vthunder_repo.get_vthunder_from_lb(
```

Comment:
Why can't we simply return vthunder here instead of calculating backup_vthunder here again?

Reply:
- The input vthunder is the current Master vthunder
- The backup_vthunder which got from get_backup_vthunder_from_lb() is the "BACKUP" vthunder in DB.
But "backup_vthunder.ip_address == vthunder.ip_address", so the "BACKUP" vthunder in DB is the current Master.
So, the "MASTER" vthunder in DB should be the current Blade. otherwise in our flow a10constants.VTHUNDER and a10constants.BACKUP_VTHUNDER will be the same instance and flow will not operate on blade when needed.